### PR TITLE
feat: docker-compose file for easy setup

### PR DESCRIPTION
**Pull request by flechtnergalani**
## relates to KILTProtocol/ticket#415

run `docker-compose up` to spin up a complete, containered development setup.
Changes in `/src` will be applied immediately, no need for a restart. Add with the `-d` flag to run in detached mode (background process, no log messages to console).
Stop and purge with `docker-compose down -v`.
A rebuild is necessary only if files outside of `/src` change (including dependencies). Trigger a rebuild with `docker-compose up --build`.

## How to test:
clone the repo in a fresh directory, start docker and run
`docker-compose up`
You should then be able to access the demo-client on `localhost:3001`

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+# instead (in this example, you could access the client on <http://localhost:1234/>).
+# You can also build and run the demo-client directly on your machine but still connect
+# to containerized prototype-services and mashnet-node:
+# ```
+# docker-compose up services mashnet-node
+# yarn install
+# yarn start
+# ```
+# In any scenario, use `docker-compose down -v` for teardown, which removes containers
+# and purges the data volumes, resetting all state.
```
Comment (tjwelde) [inline docker-compose.yml:21]: Really nice docs here!

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      - REACT_APP_SERVICE_HOST=//127.0.0.1
+      - REACT_APP_SERVICE_PORT=3000
+      - REACT_APP_NODE_HOST=127.0.0.1
+      - REACT_APP_NODE_WS_PORT=9944
+      - REACT_APP_FAUCET_URL=http://127.0.0.1:3002
+      - PUBLIC_URL=.
+    ports:
+      - 3001:3001
+    command: yarn start
+    links:
```
Comment (tjwelde) [inline docker-compose.yml:71]: links seem to be deprecated. https://docs.docker.com/compose/compose-file/#links
The other services should be reachable via service name without defining this.

Comment (flechtnergalani) [reply docker-compose.yml:71]: ah damn! was just using links to make sure that necessary services are started when starting one. Will use `depends_on` instead

**Comment by flechtnergalani** [comment]
> The `volume` prop of the demo-client doesn't seem to work correctly. I would expect, when I change a file locally, that I can see the changes in the demo-client, but I only see It on restart

That should work for anything in `src/` (just checked it again). Did you change something else? Should I try mounting the root?

**Comment by flechtnergalani** [comment]
@tjwelde We have not been able to replicate the watcher issues from your system. What was your solution again ? Will that be a too heavy penalty given that it seems to occur only infrequently?

**Pull request merged**